File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/dao/LatestVideoDao.java
package com.killrvideo.service.video.dao;

import com.datastax.oss.driver.api.mapper.annotations.Dao;
import com.datastax.oss.driver.api.mapper.annotations.Insert;
import com.datastax.oss.driver.api.mapper.annotations.StatementAttributes;
import com.killrvideo.service.video.dto.LatestVideo;

import java.util.concurrent.CompletableFuture;

@Dao
public interface LatestVideoDao {
    @Insert
    @StatementAttributes(consistencyLevel = "LOCAL_QUORUM")
    CompletableFuture<Void> insert(LatestVideo latestVideo);
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/dao/LatestVideoDao.kt
package com.killrvideo.service.video.dao

import com.datastax.oss.driver.api.mapper.annotations.Dao
import com.datastax.oss.driver.api.mapper.annotations.Insert
import com.datastax.oss.driver.api.mapper.annotations.StatementAttributes
import com.killrvideo.service.video.dto.LatestVideo
import java.util.concurrent.CompletableFuture

@Dao
interface LatestVideoDao {
    @Insert
    @StatementAttributes(consistencyLevel = "LOCAL_QUORUM")
    fun insert(latestVideo: LatestVideo): CompletableFuture<Void>
}
